lib/plugins/inspec-suggest/test/functional/suggtool_build_test.rb

```diff
@@ -0,0 +1,96 @@
+
+      # If fetch works: it must be .git directory
+      _(run_suggtool_build.stdout).must_include "fetch: working on set test-fixture"
+      _(run_suggtool_build.stdout).must_include "fetch: checked out main"
+      _(File.exist?(File.join(workdir, "test-fixture", "inspec-test-profile-basic", ".git"))).must_equal true
+
+      # If check works: check stdout for "check: Saw x error(s), y warning(s)"
+      _(run_suggtool_build.stdout).must_include "check: Saw 0 error(s), 0 warning(s)"
+
+      # If package works: inspec.yml and control/criteria.rb should be present
+      _(run_suggtool_build.stdout).must_include "package: Extracted inspec.yml from etc/suggest/test-fixture.iaf"
+      new_profile_dir = File.join(workdir, "profiles", "test-fixture")
+      inspec_yml_path = File.join(new_profile_dir, "inspec.yml")
+      control_file_path = File.join(new_profile_dir, "controls", "criteria.rb")
+      _(File.exist?(inspec_yml_path)).must_equal true
+      _(File.exist?(control_file_path)).must_equal true
+
+      # If bump works: check stdout for "bump: Bumped LEVEL version of"
+      _(run_suggtool_build.stdout).must_include "bump: Bumped patch version of"
+
+      # If sign works: verify there must be iaf file
+      _(run_suggtool_build.stdout).must_include "sign: signed 'test-fixture.iaf' OK"
+      iaf_file = File.join(outdir, "test-fixture.iaf") # Note: no version in the filename
+      _(File.exist?(iaf_file)).must_equal true
+
+      # Cleanup keys
+      FileUtils.rm(File.join(keydir, "#{keyname}.pem.key"))
+      FileUtils.rm(File.join(keydir, "#{keyname}.pem.pub"))
+    end
+  end
+
+  it "executes inspec suggtool build with failure at fetch" do
+    Dir.mktmpdir do |workdir|
+      cmd = "suggtool build non-existing-fixture"
+      cmd += " --suggest-config #{config}"
+      cmd += " --work-dir #{workdir}"
+      run_suggtool_build = run_inspec_process(cmd)
+
+      _(run_suggtool_build.exit_status).must_equal 1
+      _(run_suggtool_build.stdout).must_include "fetch: working on set non-existing-fixture"
+      _(run_suggtool_build.stdout).must_include "fetch: failed to clone https://github.com/inspec/inspec-non-existing-fixture.git"
+      _(run_suggtool_build.stdout).must_include "Repository not found"
+      _(run_suggtool_build.stdout).wont_include "check: Saw 0 error(s), 0 warning(s)"
+    end
+  end
+
+  it "executes inspec suggtool build with failure at check" do
+    Dir.mktmpdir do |workdir|
+      cmd = "suggtool build bad-fixture-branch"
+      cmd += " --suggest-config #{config}"
+      cmd += " --work-dir #{workdir}"
+      run_suggtool_build = run_inspec_process(cmd)
+
+      _(run_suggtool_build.exit_status).must_equal 1
+      _(run_suggtool_build.stdout).must_include "fetch: working on set bad-fixture-branch"
+      _(run_suggtool_build.stdout).must_include "fetch: checked out ss/profiles-for-build"
+      _(run_suggtool_build.stdout).must_include "check: Saw 2 error(s), 2 warning(s)"
```

Comment:
Is it a good idea to just check for `check:` instead of the whole string after? Because what if it does not match the exact error message and it throws some other kind of message in return.

Reply:
> It holds true for all the other test cases as well

Do you mean the strings
`"fetch: working on set bad-fixture-branch"` should be changed to `"fetch:"`
`"bump: Bumped patch version of"` should be changed to `"bump:"`
`"package: Extracted inspec.yml from"` should be changed to `"package:"`
and so on?

Reply:
Done.